### DISCUSSION TITLE
Cache is not explorable from ROOT_QUERY in dev tools version 3.0

### DISCUSSION THREAD

**Discussion by martdavidson** [apollographql/apollo-client-devtools#492] [Ideas]
This might be the desired new behaviour (in which case, this can be a feature request), but as it's a change from 2.0 and mentioned in #283 as being fixed with this new version, I figured I'd report it.

**Intended outcome:**
ROOT_QUERY cache should show full entity objects.

**Actual outcome:**
Looks like only the __ref is available in the tree, forcing you to go look for that entity in the sidebar.

![image](https://user-images.githubusercontent.com/8870176/111207407-5a3ed800-85a8-11eb-903c-c6a7e59deb9a.png)

**How to reproduce the issue:**
Open the Apollo Dev Tools on a page that makes queries, and check out the ROOT_QUERY in the cache tab. You'll see only __ref: `Entity:Id` instead of the full entity object as it was in 2.0.

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.7
 - Browser: Chrome
 - Browser version: Version 89.0.4389.82
 - Extension version: 3.0

**Comment by hwillson** [comment]
Thanks @martdavidson - this is an intentional change. We wanted to make sure the cache panel shows a real representation of the cache behind the scenes, instead of hiding ref's. This is intended to help people understand how the cache works internally. We are thinking about making this an option though, so people can toggle the behavior they prefer (either show the refs or automatically render referenced entities in the view).

**Reply by manuphatak** [reply]
Hey this is a good point.  And I recommend the dev tools to my peers for the reason -- to help them understand how the cache works!

Is there a thread anywhere where people are talking about cache 1) searchable or 2) make these ref objects clickable?  My app has 1000s of cache entries and we use UUIDs, so it can be very difficult to find your cache entry in the list.  In other words, the computer might be getting O(1) lookups, but I, as a human person, am getting O(n) lookups 😂

**Comment by martdavidson** [comment]
Understood! 

I'll give the ol' +1 to a configurable option, as you can imagine it's painful to look through a list of entities in the ROOT_QUERY and then have to cross reference the cache key to check each reference.

Anyway congrats on the launch, up to you if you want to leave this open or not.

**Comment by sasha240100** [comment]
@hwillson any updates on this topic?
